lib/onebox/engine/hackernews_onebox.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Onebox
+  module Engine
+    class HackernewsOnebox
+      include Engine
+      include LayoutSupport
+      include JSON
+
+      REGEX = /^https?:\/\/news\.ycombinator\.com\/item\?id=(?<item_id>\d+)/
+
+      matches_regexp(REGEX)
+
+      def url
+        "https://hacker-news.firebaseio.com/v0/item/#{match[:item_id]}.json"
```

Comment:
is this an official endpoint, seems a bit odd to have this decoupled.

Reply:
It's their official API that every app uses:

https://github.com/HackerNews/API